### ISSUE TITLE
deep validation for deep namespaces

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#119]
Given:
```js
// foo.js
export const x = "x"
           , y = "y"

// bar.js
export * as foo from './foo'
```

report for the following:

```js
import * as bar from './bar'
console.log(bar.foo.z) // "z non-existent in deep import './foo.js'"

import { foo } from './bar'
console.log(foo.w) // same deal
```

- [ ] report deep file with simple path relative to current source, if relative paths all the way down
  i.e. `./foo.js` as above, not `/Users/ben/code/project/src/foo.js`

**Comment by rhys-vdw** [comment]
:+1: This would be really great. If you're not planning on doing this soon I'm willing to take a look.

**Comment by benmosher** [comment]
I've got some time the next couple of days, I'll dive in and see what I can do 😄
